### ISSUE TITLE
CMake build produces different SOVERSION tags than automake

### ISSUE THREAD

**Issue by Lastique** [gperftools/gperftools#1421]
CMake build produces libraries with SOVERSION tags exactly as specified in CMakeLists.txt. For example, for `TCMALLOC_SO_VERSION` equal to 9.13.5 the resulting library is named `libtcmalloc.so.9.13.5`.

With automake, for `TCMALLOC_SO_VERSION` equal to 9:13:5 the resulting library name is `libtcmalloc.so.4.5.13`.

This is because automake interprets the specified version components and calculates SOVERSION according to [this](https://www.sourceware.org/autobook/autobook/autobook_61.html) while CMake uses the specified SOVERSION verbatim.

Both build systems should produce equivalent binaries, including the same SOVERSION.

**Comment by alk** [comment]
Correct. And thanks for taking time to file the ticket. Do note, however, that our cmake support is incomplete and best effort.
